src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useContext } from 'react';
+import { TodosFilter } from '../TodosFilter/TodosFilter';
+import { Filter } from '../../types/Filter';
+import { TodoContext, TodoUpdateContext } from '../TodosProvider/TodosProvider';
+
+interface Props {
+  changeFilter:(filter: Filter) => void;
+  currentFilter: Filter;
+}
+
+export const Footer: React.FC<Props> = (props) => {
+  const { changeFilter, currentFilter } = props;
+
+  const todos = useContext(TodoContext);
+
+  const { removeAllCompleted } = useContext(TodoUpdateContext);
+
+  const uncompletedTodos = todos.reduce((count, todo) => {
+    return todo.completed ? count : count + 1;
+  }, 0);
+  const hasCompleted = todos.some(todo => todo.completed);
+
+  return (
+    <footer className="footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${uncompletedTodos} items left`}
+      </span>
+
+      <TodosFilter
+        changeFilter={changeFilter}
+        currentFilter={currentFilter}
+      />
+
+      {hasCompleted && (
```

Comment:
```suggestion
      {!!hasCompleted && (
```

Reply:
it is already boolean as result of Array.some().

Reply:
Now - yes. But you don`t know how this condition will update in the future.